File: frontend/src/kryssCalc.ts
/**
 * Compute kryss count from minutes late (mirrors backend formula).
 *
 * raw = ln(1 + minutesLate)
 * if raw < 1 → 1
 * else standard 0.5‑up rounding to nearest int
 */
export function computeKryssForMinutes(minutesLate: number): number {
  if (minutesLate < 0) return 1;
  const raw = Math.log(1 + minutesLate);
  if (raw < 1) return 1;
  const intPart = Math.floor(raw);
  const frac = raw - intPart;
  return frac >= 0.5 ? intPart + 1 : intPart;
}

File: backend/app/kryss_calc.py
"""Kryss‑count calculation from minutes late.

Formula
-------
raw = ln(1 + minutesLate)

Rounding
--------
- If raw < 1  →  1
- Else: standard rounding with 0.5 going **up**:
    integer_part = floor(raw)
    frac = raw − integer_part
    if frac >= 0.5 → integer_part + 1
    else           → integer_part
"""

from __future__ import annotations

import math


def compute_kryss_for_minutes(minutes_late: int) -> int:
    """Return the number of kryss for a given *minutesLate* value."""
    if minutes_late < 0:
        raise ValueError("minutesLate must be >= 0")

    raw = math.log(1 + minutes_late)

    if raw < 1:
        return 1

    integer_part = math.floor(raw)
    frac = raw - integer_part

    if frac >= 0.5:
        return integer_part + 1
    return integer_part
